spot_wrapper/tests/test_graph_nav_util.py

```diff
@@ -0,0 +1,184 @@
+        )
+        # Test multiple waypoints with the same short code
+        self.graph.waypoints.add(id="AB-CD-EF-1")
+        assert (
+            graph_nav_util._find_unique_waypoint_id(
+                "AC", self.graph, self.name_to_id, self.logger
+            )
+            == "AC"
+        )
+
+
+class TestGraphNavUtilUpdateWaypointsEdges:
+    def test_empty_graph(self):
+        self.logger = logging.Logger("test_graph_nav_util", level=logging.INFO)
+
+        # Test empty graph
+        self.graph = map_pb2.Graph()
+        self.localization_id = ""
+        graph_nav_util._update_waypoints_and_edges(
+            self.graph, self.localization_id, self.logger
+        )
+        assert len(self.graph.waypoints) == 0
+        assert len(self.graph.edges) == 0
+
+    def test_one_waypoint(self):
+        self.logger = logging.Logger("test_graph_nav_util", level=logging.INFO)
+
+        # Test graph with 1 waypoint
+        self.localization_id = ""
+        self.graph = map_pb2.Graph()
+        new_waypoint = map_pb2.Waypoint()
+        new_waypoint.id = "ABCDE"
+        new_waypoint.annotations.name = "Node1"
+        self.graph.waypoints.add(
+            id=new_waypoint.id, annotations=new_waypoint.annotations
+        )
+        self.name_to_id, self.edges = graph_nav_util._update_waypoints_and_edges(
+            self.graph, self.localization_id, self.logger
+        )
+        assert len(self.graph.waypoints) == 1
+        assert len(self.graph.edges) == 0
+        assert len(self.edges) == 0
+        assert len(self.name_to_id) == 1
+        assert self.name_to_id["Node1"] == "ABCDE"
+
+    def test_two_waypoints_with_edge(self):
+        self.logger = logging.Logger("test_graph_nav_util", level=logging.INFO)
+
+        # Test graph with 2 waypoints and an edge between them
+        self.localization_id = ""
+        self.graph = map_pb2.Graph()
+        new_waypoint = map_pb2.Waypoint()
+        new_waypoint.id = "ABCDE"
+        new_waypoint.annotations.name = "Node1"
+        self.graph.waypoints.add(
+            id=new_waypoint.id, annotations=new_waypoint.annotations
+        )
+        new_waypoint.id = "DE"
+        new_waypoint.annotations.name = "Node2"
+        self.graph.waypoints.add(
+            id=new_waypoint.id, annotations=new_waypoint.annotations
+        )
+        new_edge = map_pb2.Edge.Id(from_waypoint="ABCDE", to_waypoint="DE")
+
+        self.graph.edges.add(id=new_edge)
+        self.name_to_id, self.edges = graph_nav_util._update_waypoints_and_edges(
+            self.graph, self.localization_id, self.logger
+        )
+        assert len(self.graph.waypoints) == 2
+        assert len(self.graph.edges) == 1
+        assert len(self.edges) == 1
+        assert self.edges["DE"][0] == "ABCDE"
+        assert len(self.name_to_id) == 2
+        assert self.name_to_id["Node1"] == "ABCDE"
+        assert self.name_to_id["Node2"] == "DE"
+
+    def test_two_waypoints_with_edge_and_localization(self):
```

Comment:
Is there a difference between this test and the test above? I'm not as familiar with this code so maybe I'm missing something.